src/loaders/core_text.rs

```diff
@@ -122,29 +122,14 @@ impl Font {
     }
 
     /// Creates a font from a native API handle.
-    pub unsafe fn from_native_font(core_text_font: NativeFont) -> Font {
-        Font::from_core_text_font(core_text_font)
-    }
-
-    unsafe fn from_core_text_font(core_text_font: NativeFont) -> Font {
-        let mut font_data = FontData::Unavailable;
-        match core_text_font.url() {
-            None => warn!("No URL found for Core Text font!"),
-            Some(url) => match url.to_path() {
-                Some(path) => match File::open(path) {
-                    Ok(ref mut file) => match utils::slurp_file(file) {
-                        Ok(data) => font_data = FontData::Memory(Arc::new(data)),
-                        Err(_) => warn!("Couldn't read file data for Core Text font!"),
-                    },
-                    Err(_) => warn!("Could not open file for Core Text font!"),
-                },
-                None => warn!("Could not convert URL from Core Text font to path!"),
-            },
-        }
-
+    pub unsafe fn from_native_font(core_text_font: &NativeFont) -> Font {
+        Font::from_core_text_font_no_path(core_text_font.clone())
+    }
+    /// Creates a font from a native API handle, without performing a lookup on the disk.
+    pub unsafe fn from_core_text_font_no_path(core_text_font: NativeFont) -> Font {
         Font {
             core_text_font,
-            font_data,
+            font_data: FontData::Unavailable,
```

Comment:
Ha, good catch; `Font` implements Deref to [u8] which will panic if there's no data associated: https://github.com/servo/font-kit/blob/master/src/loaders/core_text.rs#L784
Granted, this can still happen on master if a font has no path.
Also, `Font::copy_font_data` will return None instead of font contents, but that at least does not panic in surprising ways. I'd say that if the panic and implicit deref to [u8] poses a problem, one can always turn to using `Font::copy_font_data`, which gives a chance to avoid a panic.

Reply:
On the other hand, this PR is already Semver-breaking with a change to `Loader::from_native_font`, so we might as well get rid of the Deref implementation?